core/src/main/java/org/infinispan/util/CacheInjectable.java

```diff
@@ -0,0 +1,15 @@
+package org.infinispan.util;
+
+import org.infinispan.Cache;
+
+/**
+ * Interface that describes how a cache can be injected into another object.  This is useful for cases such as
+ * after an object is deserialized and you must inject a Cache into it.
+ */
+public interface CacheInjectable<K, V> {
```

Comment:
I can change it over if you wanted.  We have used *Aware in quite a few places and they seem to have different connotations where as Injectable seemed clear to me, but I guess not for others.